polyfill/test/PlainYearMonth/constructor/constructor/infinity-throws-rangeerror.js

```diff
@@ -13,17 +14,33 @@ assert.throws(RangeError, () => new Temporal.PlainYearMonth(Infinity, 1));
 assert.throws(RangeError, () => new Temporal.PlainYearMonth(1970, Infinity));
 assert.throws(RangeError, () => new Temporal.PlainYearMonth(1970, 1, isoCalendar, Infinity));
 
-let calls = 0;
-const obj = {
-  valueOf() {
-    calls++;
-    return Infinity;
-  }
-};
+const O = (primitiveValue, propertyName) => (calls) => TemporalHelpers.toPrimitiveObserver(calls, primitiveValue, propertyName);
+const tests = [
+  [
+    "infinite year",
+    [O(Infinity, "year"), O(1, "month"), () => "iso8601", O(1, "day")],
+    ["get year.valueOf", "call year.valueOf"]
+  ],
+  [
+    "infinite month",
+    [O(1970, "year"), O(Infinity, "month"), () => "iso8601", O(1, "day")],
+    ["get year.valueOf", "call year.valueOf", "get month.valueOf", "call month.valueOf"]
+  ],
+  [
```

Comment:
This invalid-calendar test is a bit off-topic for this test file, though I could also see the reasoning for including it here. It's essentially duplicated in the negative-infinity test though. I don't feel strongly about it.